packages/server/api/controllers/orders.controller.js

```diff
@@ -33,6 +33,83 @@ const getOrdersByUserId = async (token) => {
   }
 };
 
+// get order with status "created" by user-id
+const addOrderItemsByUserId = async (authorization, body) => {
+  try {
+    const userUid = authorization.split(' ')[1];
+    const user = (await knex('users').where({ uid: userUid }))[0];
+    if (!user) {
+      throw new HttpError('User not found', 401);
+    }
+
+    const { product_id: productId, color, size, quantity } = body;
+
+    if (!productId && !color && !size && !quantity) {
+      throw new HttpError('Wrong product parameters', 400);
+    }
+
+    let [order] = await knex('orders').where({
+      user_id: user.id,
+      status: 'created',
+    });
+
+    if (!order) {
+      const orderId = await knex('orders').insert({
+        user_id: user.id,
+        status: 'created',
+      });
+
+      [order] = await knex('orders').where({
+        id: orderId,
+      });
+    }
+
+    const [variant] = await knex('variants').where({
+      product_id: productId,
+      color,
+      size,
+    });
+
+    if (!variant) {
+      throw new HttpError('Variant is not available', 404);
+    }
+
+    const [productsInQueue] = await knex
+      .select(
+        'order_items.variant_id',
+        knex.raw('SUM(order_items.quantity) as quantity'),
+      )
+      .from('order_items')
+      .where({
+        order_id: order.id,
+        variant_id: variant.id,
+      })
+      .groupBy('order_items.variant_id');
+
+    const availableProducts = variant.stock - productsInQueue.quantity;
+
+    if (quantity > availableProducts) {
+      throw new HttpError(
+        'Order products quantity is bigger than stock quantity',
+        400,
+      );
+    } else {
+      await knex('order_items').insert({
```

Comment:
If I add the same variant multiple times, I would get multiple order items for the same variant. The behavior I would expect to see is that existing order items are updated. I don't think your approach is incorrect though but just mentioning as a potential improvement.

Reply:
Thank, I though about that also. I have improved `get order items` in other PR. So query just groups all of `order items` with the same `order_id` and `variants_id` together. 

But yes, update is much more easy and maybe lighter.

Reply:
I though about that, but so some reasons decided to group order items with the same `order_id` and `variant_id` together in other PR, that is pelated to ShoppinCartContext. Updating must be more efficient.